sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -536,6 +538,223 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
     });
   });
 
+  it("Should sum the grants from past applications with different offering intensities when the applications are for the same student and program year.", async () => {
+    // Arrange
+
+    // Create the student and program year to be shared across the applications.
+    const student = await saveFakeStudent(db.dataSource);
+    // Get the program year for the start date.
+    const programYear = await db.programYear.findOne({ where: { id: 2 } });
+
+    // Past part-time application with federal and provincial loans and grants.
+    // Loans will be ignored.
+    await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        student,
+        firstDisbursementValues: [
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaLoan,
+            "CSLP",
+            1,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSPT",
+            2,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSGD",
+            3,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSGP",
+            4,
+          ),
+          // Should not be disbursed due to BCLM restriction.
+          createFakeDisbursementValue(DisbursementValueType.BCGrant, "SBSD", 6),
+          // Should not be disbursed due to BCLM restriction.
+          createFakeDisbursementValue(DisbursementValueType.BCGrant, "BCAG", 7),
```

Comment:
The comments about restrictions are probably no longer required.
The test can be more focused in the workflow totals, so you don't need for all these grants.